scripts/ci/test_pr_governance_gate.sh

```
@@ -0,0 +1,264 @@
+  DIRECT_PR_NUMBER="" \
+  WORKFLOW_RUN_PR_NUMBER="" \
+    bash "$script" > "$temp_dir/output.txt"
+}
+
+assert_no_comment_or_merge_for_pending_checks() {
+  local temp_dir
+  temp_dir="$(mktemp -d)"
+  run_gate pending "$temp_dir"
+
+  grep -q 'Waiting for 1 required check' "$temp_dir/output.txt"
+  ! grep -q 'issues/42/comments -f body' "$temp_dir/gh.log"
```

Shellcheck (0.11.0)

[info] This ! is not on a condition and skips errexit. Use `&& exit 1` instead, or make sure $? is checked.

(SC2251)